libhb/preset.c

```diff
@@ -165,6 +170,50 @@ static int do_preset_search(hb_value_t *preset, preset_do_context_t *do_ctx)
     return result;
 }
 
+static int preset_hw_scrub(hb_value_t *preset)
+{
+    int disabled = 0;
+    hb_value_t *val = hb_dict_get(preset, "VideoEncoder");
+    if (val != NULL)
+    {
+        const char *s;
+        int vcodec;
+        s = hb_value_get_string(val);
+        vcodec = hb_video_encoder_get_from_name(s);
+        if (vcodec != HB_VCODEC_INVALID)
+        {
+            if (vcodec & HB_VCODEC_QSV_MASK)
+            {
+                if(hb_qsv_available())
+                {
+#if HB_PROJECT_FEATURE_QSV
+                    // check the qsv codec is supported by hw
+                    disabled = hb_qsv_video_encoder_is_enabled(vcodec) ? 0 : 1;
+#endif
+                }
+                else
+                {
+                    disabled = 1;
+                }
+            }
+            // TODO: other hw codecs for non Intel platforms
+        }
+    }
+
+    if(disabled)
+    {
+        hb_dict_set_int(preset, "PresetDisabled", disabled);
```

Comment:
This new key should be added to preset/preset_template.json with a value of false.  The template is used for validation of preset keys and value types.  So if it's not in there, some code now or in the future might scrub the instances you add here.

Reply:
Done in e63bbdfa58f5989dbe2675b0637c119ef96dbe8c.